# Bioinformatics Project: Differential Expression Analysis using Python and R

## Project Description

This project aims to identify differentially expressed genes between two conditions (e.g., healthy vs diseased) using RNA-seq data. The project consists of five steps:

- Quality control and trimming of FASTQ files using Python tools
- Alignment of reads to a reference genome using Python tools
- Quantification of gene expression levels using Python tools
- Differential expression analysis using R packages
- Visualization and exploration of results using R packages

## Project Goal

The goal of this project is to learn how to use Python and R tools for bioinformatics analysis, and to gain insights into the biological mechanisms underlying the differences between the two conditions.

## Skills Needed for Each Step

- For step one (quality control and trimming of FASTQ files), you need to have basic Python skills such as reading and writing files, using loops and conditional statements, and importing libraries. You also need to know how to use command line arguments or configuration files to run Python scripts or packages. Some of the libraries and tools that you can use for this step are FastQC, Trimmomatic, FASTQ_tools, fastq-filter, BioPython and PySam.
- For step two (alignment of reads to a reference genome), you need to have basic Python skills such as reading and writing files, using loops and conditional statements, and importing libraries. You also need to know how to use command line arguments or configuration files to run Python scripts or packages. Some of the libraries and tools that you can use for this step are HISAT2, STAR, BioPython and PySam.
- For step three (quantification of gene expression levels), you need to have basic Python skills such as reading and writing files, using loops and conditional statements, and importing libraries. You also need to know how to use command line arguments or configuration files to run Python scripts or packages. Some of the libraries and tools that you can use for this step are featureCounts, HTSeq, BioPython and PySam.
- For step four (differential expression analysis), you need to have basic R skills such as reading and writing data frames, using loops and conditional statements,
applying functions,
performing statistical tests,
and importing packages.
You also need
to know how
to use RStudio or other IDEs for R programming.
Some of the packages that you can use for this step are edgeR or DESeq2.
- For step five (visualization
and exploration
of results),
you need
to have basic R skills such as reading
and writing data frames,
using loops
and conditional statements,
applying functions,
creating plots,
adding annotations,
customizing aesthetics,
saving outputs in various formats.
You also need
to know how
to use RStudio or other IDEs for R programming.
Some of the packages that you can use for this step are ggplot2,
pheatmap,
gplots or clusterProfiler.

## How To Run The Project

To run the project,
you need
to download
the input data,
the code files,
and the output folder from GitHub.
You also need
to install
the required libraries
and tools
for each step.
Then,
you can follow
the instructions in each code file
to execute them in order.
The output folder will contain
the results from each step,
such as trimmed FASTQ files,
aligned BAM files,
count tables,
differentially expressed genes,
and plots.

## References

: https://github.com/alexkychen/FASTQ_tools  
: https://pypi.org/project/fastq-filter/  
: https://hbctraining.github.io/Intro-to-rnaseq-hpc-salmon-flipped/lessons/06_qc_running_fastqc_sbatch.html  
: https://bioconductor.org/packages/release/bioc/html/edgeR.html  
: https://bioconductor.org/packages/release/bioc/html/DESeq2.html  
: https://ggplot2.tidyverse.org/  
: https://cran.r-project.org/web/packages/pheatmap/index.html  
: https://cran.r-project.org/web/packages/gplots/index.html  
: https://bioconductor.org/packages/release/bioc/html/clusterProfiler.html  